src/MagicString.js

```diff
@@ -780,4 +780,25 @@ export default class MagicString {
 		}
 		return this;
 	}
+
+	_replaceString(string, replacement) {
+		const { original } = this;
+		const stringLength = string.length;
+		for (
+			let index = original.indexOf(string);
+			index !== -1;
+			index = original.indexOf(string, index + stringLength)
+		) {
+			this.overwrite(index, index + stringLength, replacement);
+		}
+
+		return this;
+	}
+
+	replace(searchValue, replacement) {
+		return this[typeof searchValue === 'string' ? '_replaceString' : '_replaceRegexp'](
```

Comment:
Can we use an if condition instead of this? Thanks.

Reply:
Done